cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -19,6 +19,9 @@ spec:
         version: v11
         kubernetes.io/cluster-service: "true"
     spec:
+{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
```

Comment:
Without the use of host networking, attempts to resolve hostnames were slow. I tracked it down to the attempt to inherit DNS from the host: this allows us to use the site's DNS to resolve names that are not maintained by SkyDNS. 

This was because the pod wasn't able to access to access the hosts network (It eventually times out, which is why it was slow, not a failure.). Giving it access to the host network solved the problem.

I wasn't aware of it being a bad practice to use host networking and would love to know more. 

It's possible that I've misunderstood the Kubernetes networking model, and I've set things up in a way that is not correct. Should it be different? I read over http://kubernetes.io/docs/admin/networking/ and it says:

> Kubernetes imposes the following fundamental requirements on any networking implementation (barring any intentional network segmentation policies):
> * all containers can communicate with all other containers without NAT
> * all nodes can communicate with all containers (and vice-versa) without NAT
> * the IP that a container sees itself as is the same IP that others see it as

As far as I can tell, I've implemented the expected networking model... but it's also clear that I needed to set host networking and other kube-up implementations didn't, so perhaps I haven't fully understood.